app/src/pages/explorer/curate/spreadsheet/spreadsheet-upload-script.js

```diff
@@ -0,0 +1,195 @@
+      fourth: false,
+      fifth: false,
+      sixth: false,
+      isInvalidForm: false,
+      dialog: {
+        title: '',
+        type: null,
+        size: 60,
+        disableClose: false
+      },
+      navRoutes: [
+        {
+          label: 'Curate',
+          path: '/explorer/curate'
+        },
+        {
+          label: 'Spreadsheet',
+          path: '/explorer/curate/spreadsheet'
+        }
+      ]
+    }
+  },
+  props: {
+    datasetId: {
+      type: String
+    }
+  },
+  computed: {
+    ...mapGetters({
+      userId: 'auth/userId',
+      isAuthenticated: 'auth/isAuthenticated',
+      dialogBoxActive: 'dialogBox'
+    })
+  },
+  methods: {
+    addSpreadsheet: spreadsheetFn.addFiles,
+    removeSpreadsheet: spreadsheetFn.removeFile,
+    modStatSpreadsheet: spreadsheetFn.modifyStatus,
+    addSupp: suppFn.addFiles,
+    removeSupp: suppFn.removeFile,
+    modStatSupp: suppFn.modifyStatus,
+    ...mapMutations({
+      toggleDialogBox: 'setDialogBox'
+    }),
+    async navBack () {
+      this.$router.back()
+    },
+    async createDataset () {
+      await this.$apollo.mutate({
+        mutation: CREATE_DATASET_MUTATION,
+        variables: {
+          input: {
+            datasetId: `${this.datasetId}`,
+            files: this.uploadedFiles.files,
+            title: this.title,
+            doi: this.doi
+          }
+        }
+      }).then((data) => {
+        return data
+      }).catch((error) => {
+        console.error('error:', error)
+      })
+    },
+    // Format files for submission
+    processFiles () {
+      var incompleteFiles = this.spreadsheetFiles.filter(file => file.status === 'incomplete')
+        .concat(this.suppFiles.filter(file => file.status === 'incomplete'))
+      return incompleteFiles.map(({ file }) => file)
+    },
+    onInputChange (e) {
+      if (e.target.id === 'file-spreadsheet-input') {
+        this.addSpreadsheet(e.target.files)
+      } else {
+        this.addSupp(e.target.files)
+      }
+      // reset so that selecting the same file again will still cause it to fire this change
+      e.target.value = null
+    },
+    goToStep (id, index) {
+      if (id === 'fourth' && index === 'sixth') {
+        if (!this.title) {
+          this.isInvalidForm = true
+          return
+        } else this.isInvalidForm = false
+      }
+      this[id] = true
+      if (index) {
+        this.active = index
+      }
+    },
+    async submitFiles () {
+      this.toggleDialogBox()
+      this.uploadInProgress = 'Uploading files'
+      this.renderDialog('Submitting dataset', 'loading', 40, true)
+      await this.uploadFiles()
+        .then(() => {
+          this.uploadInProgress = 'Creating dataset'
+          this.createDataset()
+          setTimeout(() => {
+            this.toggleDialogBox()
+            this.uploadInProgress = false
+            this.$router.push({ name: 'DatasetSingleView', params: { id: `${this.datasetId}` } })
+          }, 1000)
+        })
```

Comment:
Should we try the try and catch and get rid of the `.then`? Not sure if we need to `catch` anything here since you don't have a `catch block` any way. Just avoid mixing await and then. Same everywhere it applies.

Reply:
I'm changing this one to a try/catch